driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -572,6 +638,14 @@ public ServerDescription getInitialServerDescription() {
         }
     }
 
+    private static final class MongoOpenConnectionInternalException extends RuntimeException {
```

Comment:
Nit: at it's not so common in the driver to have a non-public exception type, consider a short Javadoc comment explaining its purpose.

Reply:
Done.